### ISSUE TITLE
maven_jar skylark test is broken 

### ISSUE THREAD

**Issue by hanwen** [bazelbuild/bazel#2086]
see  below. test_maven_aar_skylark should do something so the API level for android NDK/SDK is set in the workspace.
```
hanwen@hanwen:~/vc/bazel$ git show HEAD |head -1
commit 18033b0996e96faf67536b43165b3f40f234a3f7

hanwen@hanwen:~/vc/bazel$ git diff 
diff --git a/WORKSPACE b/WORKSPACE
index cace908..d71af7a 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -56,5 +56,5 @@ bind(name = "proto/toolchains/java", actual = "//third_party/protobuf:java_toolc
 
 # For Skylark tests at //src/test/shell/bazel:maven_skylark_test
 # Uncomment the following lines, and the test in src/test/shell/bazel/BUILD to run it.
-# load("//tools/build_defs/repo:maven_rules.bzl", "maven_dependency_plugin")
-# maven_dependency_plugin()
+load("//tools/build_defs/repo:maven_rules.bzl", "maven_dependency_plugin")
+maven_dependency_plugin()
diff --git a/src/test/shell/bazel/maven_skylark_test.sh b/src/test/shell/bazel/maven_skylark_test.sh
index e1d1f48..7b4d7b4 100755
--- a/src/test/shell/bazel/maven_skylark_test.sh
+++ b/src/test/shell/bazel/maven_skylark_test.sh
@@ -132,6 +132,8 @@ maven_aar(
     settings = "$local_maven_settings_xml",
 )
 EOF
+  cat WORKSPACE
+  echo "hoi"
   bazel build //java/com/app || fail "Expected build to succeed"
   unzip -l bazel-bin/java/com/app/app.apk > $TEST_log
   expect_log_once "res/layout/my_view.xml"

$ bazel test --test_output=streamed --test_filter=test_maven_aar_skylark --spawn_strategy=standalone  src/test/shell/bazel:maven_skylark_test
...

/home/hanwen/.cache/bazel/_bazel_hanwen/0c4198804daf70e6b39cbe31e3a971c4/execroot/bazel/_tmp/maven_skylark_test_1/workspace
workspace(name = 'main')
android_ndk_repository(
    name = "androidndk",
    path = "/home/hanwen/.cache/bazel/_bazel_hanwen/0c4198804daf70e6b39cbe31e3a971c4/execroot/bazel/_tmp/maven_skylark_test_1/workspace/android_ndk",
    api_level = ,
)
android_sdk_repository(
    name = "androidsdk",
    path = "/home/hanwen/.cache/bazel/_bazel_hanwen/0c4198804daf70e6b39cbe31e3a971c4/execroot/bazel/_tmp/maven_skylark_test_1/workspace/android_sdk",
    build_tools_version = "",
    api_level = ,
)
load("@bazel_tools//tools/build_defs/repo:maven_rules.bzl", "maven_aar")
maven_aar(
    name = "herbivore",
    artifact = "com.example.carnivore:herbivore:1.21",
    sha1 = "d00cb11b900e4df82e6a3fcba1624e29f55d80b4",
    settings = "/home/hanwen/.cache/bazel/_bazel_hanwen/0c4198804daf70e6b39cbe31e3a971c4/execroot/bazel/_tmp/maven_skylark_test_1/workspace/settings.xml",
)
hoi
INFO: Reading 'startup' options from /home/hanwen/.cache/bazel/_bazel_hanwen/0c4198804daf70e6b39cbe31e3a971c4/execroot/bazel/_tmp/maven_skylark_test_1/bazelrc: --output_user_root=/home/hanwen/.cache/bazel/_bazel_hanwen/0c4198804daf70e6b39cbe31e3a971c4/execroot/bazel/_tmp/maven_skylark_test_1/root --host_javabase=/home/hanwen/.cache/bazel/_bazel_hanwen/0c4198804daf70e6b39cbe31e3a971c4/execroot/bazel/bazel-out/local-fastbuild/bin/src/test/shell/bazel/maven_skylark_test.runfiles/local_jdk
INFO: $TEST_TMPDIR defined: output root default is '/home/hanwen/.cache/bazel/_bazel_hanwen/0c4198804daf70e6b39cbe31e3a971c4/execroot/bazel/_tmp/maven_skylark_test_1'.
ERROR: /home/hanwen/.cache/bazel/_bazel_hanwen/0c4198804daf70e6b39cbe31e3a971c4/execroot/bazel/_tmp/maven_skylark_test_1/workspace/WORKSPACE:5:17: syntax error at ',': expected expression.
ERROR: /home/hanwen/.cache/bazel/_bazel_hanwen/0c4198804daf70e6b39cbe31e3a971c4/execroot/bazel/_tmp/maven_skylark_test_1/workspace/WORKSPACE:12:17: syntax error at ',': expected expression.
ERROR: error loading package 'external': Failed to parse WORKSPACE file.
____Elapsed time: 0.195s
```

**Comment by aj-michael** [comment]
Ah shucks, test_maven_aar_skylark failed because there is no android_sdk_repository in your in your WORKSPACE. Unfortunately that comment about "Uncomment the following lines" is no longer sufficient to run this test, because test_maven_aar_skylark is essentially an android integration test, so it requires filling in the android_sdk_for_testing bind rule in the WORKSPACE.

Although, once I add that, it unearths another problem with test_maven_aar_skylark with regards to a change in the default Android manifest merger last week that was not caught because this test is marked manual and does not run on CI.

Thanks for filing this issue!

**Comment by aj-michael** [comment]
Fixed by 8a136d8 and f16f7c5.
